File: integer_to_roman.py
rtoi = {
  "M": 1000,
  "CM": 900,
  "D": 500,
  "CD": 400,
  "C": 100,
  "XC": 90,
  "L": 50,
  "XL": 40,
  "X": 10,
  "IX": 9,
  "V": 5,
  "IV": 4,
  "I": 1  
}

class Solution:
    def intToRoman(self, num: int) -> str:
        ans = ""
        for key, value in rtoi.items():
          if value <= num:
            quo = num // value
            for i in range(quo):
              ans += key
            num -= quo * value
        return ans
      
if __name__ == "__main__":
  s = Solution()
  print(s.intToRoman(111))
            
          
        

File: integer-to-roman.js
function intToRoman(num): {
  const rtoi = {
      "I": 1,
      "IV": 4,
      "V": 5,
      "X": 10,
      "XL": 40,
      "L": 50,
      "C": 100,
      "CD": 400,
      "D": 500,
      "M": 1000
  };
  
  let n = num;
  let ans = "";
  for (let key in rtoi) {
      if (rtoi[key] >= n) {
          const quotient = Math.floor(n / rtoi[key]);
          for (let i = 0; i < quotient; ++i)
          {
              ans += key;
          }
      }
  }

  return ans;
};

(() =>console.log(intToRoman(111)))();